### PR TITLE
Switch to go 1.7

### PR DESCRIPTION
**Pull request by tsg**
This also updates the OS X cross compiler used in pacakging to use the 10.11
SDK. Credit for figuring out what's needed goes to the upstream project
github.com/karalabe/xgo.

### CODE REVIEW

**Comment by tsg** [comment]
jenkins, package it

**Comment by tsg** [comment]
@ruflin i ended up doing a new PR because your #2306 was needing a rebase anyway. Can you check that I didn't miss anything that you had in it?

File: dev-tools/packer/docker/xgo-image/base/patch.tar.xz [hunk 1]
```diff
@@ -0,0 +1 @@
+<html><body>You are being <a href="https://raw.githubusercontent.com/karalabe/xgo/master/docker/base/patch.tar.xz">redirected</a>.</body></html>
```
Comment (andrewkroh) [inline dev-tools/packer/docker/xgo-image/base/patch.tar.xz]: This looks like it didn't follow a redirect on download.

Comment (tsg) [inline dev-tools/packer/docker/xgo-image/base/patch.tar.xz]: Ha, strange that it seemed to work on my computers. Ooops.

**Comment by tsg** [comment]
jenkins, package it

**Comment by tsg** [comment]
Jenkins package green now: http://build-eu-00.elastic.co/view/Beats/job/beats-package-PR/24/

**Comment by ruflin** [comment]
@tsg I think you missed some appveyor lines

**Comment by ruflin** [comment]
Ah, an update the CHANGELOG.

**Pull request merged**